ahnlich/ai/src/cli/server.rs

```diff
@@ -177,6 +213,30 @@ impl From<&SupportedModels> for Model {
     }
 }
 
+impl From<&Model> for SupportedModels {
+    fn from(value: &Model) -> Self {
+        let supported_model = match value {
+            Model::Text { name, .. } => {
+                match name.as_str() {
+                    "all-MiniLM-L6-v2" => SupportedModels::AllMiniLML6V2,
+                    "all-MiniLM-L12-v2" => SupportedModels::AllMiniLML12V2,
+                    "bge-base-en-v1.5" => SupportedModels::BGEBaseEnV15,
```

Comment:
I think this doesn't match with the name assigned in `model.rs`.... but I think generally if only there was a way to define names in just one single place as opposed to the 3 places we have right now i.e (clap-name, this conversion and AI into Model conversion)

Particularly because they're just strings and not types and so we can skip on things like equality or forgetting usecases by mistake

Reply:
To solve this I have an inkling of an approach. We can do it like this

We keep the existing `SupportedModels` with the `clap(name)` derivations. but then we implement the enum and define a kind of Display name

```rust
enum SupportedModels {
     #[clap(name = "all-mini-lm-l6-v2")]
    AllMiniLML6V2,
}

impl std::fmt::Display for SupportedModels  {
  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let name = match self {
         AllMiniLML6V2 =>. "all-MiniLM-L6-V2",         
        _ => ...
        };
        write!(f, "{}", name)
    }}
```

This way, instead of storing a name inside `Model`, we can store a `SupportedModels` and turn it into a string wherever else we need it.... it would remove string matching from this `From` statement and also in `list_supported_models`, we can get name from calling `supported_model.to_string()`